covid19-etl/etl/cityofchicago.py

```diff
@@ -0,0 +1,430 @@
+#  This ETL is for city of chicago dataset for COVID-19 Daily Cases, Deaths, and Hospitalizations (CDH)
+#  Reference: https://data.cityofchicago.org/Health-Human-Services/COVID-19-Daily-Cases-Deaths-and-Hospitalizations/naz8-j4nc
+
+import boto3
+import csv
+from datetime import datetime, timedelta
+import time
+import json
+import requests
+import pandas as pd
+from contextlib import closing
+from utils.metadata_helper import MetadataHelper
+from utils.format_helper import (
+    derived_submitter_id,
+    format_submitter_id,
+)
+from etl import base
+import sys
+
+CITYOFCHICAGO_CDH_URL = "https://data.cityofchicago.org/resource/naz8-j4nc.csv"
+
+
+def str_to_int(string):
+    # Method to parse numbers from string to int and 0 if string it empty
+    return int(string or 0)
+
+
+def str_to_datetime(string):
+    # Method to convert date from string to datetime
+    try:
+        return datetime.strptime(string, "%Y-%m-%d")
+    except ValueError:
+        raise Exception(f"ParseError: The date format is not Valid for {string}")
+
+
+class CITYOFCHICAGO(base.BaseETL):
+    def __init__(self, base_url, access_token, s3_bucket):
+        self.base_url = base_url
+        self.access_token = access_token
+        self.s3_bucket = s3_bucket
+
+        self.program_name = "open"
+        self.project_code = "cityofchicago"
+        self.metadata_helper = MetadataHelper(
+            base_url=self.base_url,
+            program_name=self.program_name,
+            project_code=self.project_code,
+            access_token=self.access_token,
+        )
+
+        self.city = "Chicago"
+        self.county = "Cook"
+        self.country = "US"
+        self.state = "IL"
+        self.last_submission_identifier = None  # last_submission_identifier is used in project node, here this identifiers marks the date for last date where hospitalization data was present
+
+        self.summary_locations = {}  # { <submitter_id>: <record> }
+        self.summary_clinicals = {}
+        self.summary_group_demographics = {}
+
+    def get_summary_location(self, summary_location_submitter_id):
+        # This dataset would only require one `summary_location` which is chicago, so the entry is made in `summary_location` only if it doesn't already exsist
+
+        current_summary_location = self.metadata_helper.get_existing_summary_locations()
+        if len(current_summary_location) > 0:
+            return
+        else:
+            self.summary_locations[summary_location_submitter_id] = {
+                "country_region": self.country,
+                "county": self.county,
+                "province_state": self.state,
+                "projects": [{"code": self.project_code}],
+            }
```

Comment:
@urvi-occ i tried to run the ETL in QA and got this error:
```
Submitting summary_location data
[...]
Exception: Unable to submit to Sheepdog: 400
{"code":400,"created_entity_count":0,"entities":[{"action":null,"errors":[{"keys":["submitter_id"],"message":"'submitter_id' is a required property","type":"MISSING_PROPERTY"}],"id":null,"type":"summary_location","unique_keys":[{"project_id":"open-cityofchicago","submitter_id":null}],"valid":false,"warnings":[]}],"entity_error_count":1,"message":"Transaction aborted due to 1 invalid entity.","success":false,"transaction_id":316584,"transactional_error_count":0,"transactional_errors":[],"updated_entity_count":0}
```

Reply:
Looks like summary_location might need submitter_id defined in the dict, I made a new commit. Please let me know if it errors out after that.

Reply:
@urvi-occ 
```
Submitting data...
Submitting summary_location data
  Submitting 1 records in batches of 100
Submission progress: 1/1
Submitting summary_clinical data
  Submitting 804 records in batches of 100

Exception: Unable to submit to Sheepdog: 400
[...] "errors":[{"keys":[],"message":"Additional properties are not allowed ('hospitaliIzedCumulative' was unexpected)","type":"ERROR"}]
```

Reply:
My bad. There was an extra I in the key `hospitalizedCumulative`. I would be removing it and making a new commit soon